progress.go

```diff
@@ -110,28 +123,32 @@ func (p *progress) isDone() (bool, error) {
 
 // removes this progress file it the download is done
 func (p *progress) close() error {
-	ok, err := p.isDone()
+	done, err := p.isDone()
 	if err != nil {
-		return fmt.Errorf("error checking if donwload is done: %w", err)
+		return fmt.Errorf("error checking if %s is done: %w", p.URL, err)
 	}
-	if !ok {
-		return nil
-	}
-	if err := os.Remove(p.path); err != nil && !os.IsNotExist(err) {
-		return fmt.Errorf("error cleaning up progress files %s: %w", p.path, err)
+	if done {
+		if err := os.Remove(p.path); err != nil && !os.IsNotExist(err) {
+			return fmt.Errorf("error cleaning up progress file %s: %w", p.path, err)
+		}
 	}
 	return nil // Either not empty or error, suits both cases
 }
 
 func newProgress(path, url string, chunkSize int64, chunks int, restart bool) (*progress, error) {
-	absPath, err := filepath.Abs(path)
+	dir, err := getChunkDirectory()
+	if err != nil {
+		return nil, fmt.Errorf("could not get chunk's directory: %w", err)
+	}
+	abs, err := filepath.Abs(path)
 	if err != nil {
-		return nil, fmt.Errorf("error getting absolute path for %s: %w", path, err)
+		return nil, fmt.Errorf("could not get the download absolute path: %w", err)
 	}
+	hash := fmt.Sprintf("%x", md5.Sum([]byte(fmt.Sprintf("%s|%s", url, abs))))
 	p := progress{
-		path:      filepath.Join(filepath.Dir(absPath), progressFilePrefix+filepath.Base(absPath)),
+		path:      filepath.Join(dir, fmt.Sprintf("%s-%s", hash, filepath.Base(path))),
```

Comment:
Could you add a comment explaining why we are adding this suffix? Thinking of myself one year from now, scratching my head to understand why.